src/app/dim-ui/usePopper.ts

```diff
@@ -160,5 +168,5 @@ export function usePopper({
     }
 
     return destroy;
-  });
+  }, [reference, contents, state]);
```

Comment:
My original idea behind having the layout effect fire on every update was that I don't know what (if anything) has changed about the contents, so I don't know when exactly to re-position. This is made especially complicated by the fact that popper seems to have some bugs around its lifecycle (which afaik are fixed in floating-ui).

Testing around a bit, I can see that uses of `Dropdown` appear broken with this change - for example, the menu you can access from the three dots in the search bar.

Reply:
Thanks for testing! I'll go back and take another look. It's likely that I missed a case I was not considering before.

Reply:
Ok I made some adjustments. The big change here is `usePopper` now accepts a dependency list. For the particular broken popovers you noted, both the `isOpen` state and the `items` list are listed as dependencies in addition to all the other deps in `usePopper`. This should resolve the problem you noted.

I opted to note touch the other `usePopper` implementations because they currently do not exhibit the reported bug.